analyzers/nodejs/integration_test.go

```diff
@@ -0,0 +1,211 @@
+package nodejs_test
+
+import (
+	"flag"
+	"os"
+	"path/filepath"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/fossas/fossa-cli/analyzers"
+	"github.com/fossas/fossa-cli/exec"
+	"github.com/fossas/fossa-cli/files"
+	"github.com/fossas/fossa-cli/module"
+	"github.com/fossas/fossa-cli/pkg"
+	"github.com/fossas/fossa-cli/testing/fixtures"
+	"github.com/fossas/fossa-cli/testing/runfossa"
+)
+
+var nodeAnalyzerFixtureDir = filepath.Join(fixtures.FixtureDirectory(), "nodejs", "analyzer")
+
+func TestMain(m *testing.M) {
+	// flags are not parsed at this point. In order to have testing.Short() read actually provided values, this must be executed
+	flag.Parse()
+	if testing.Short() {
+		return
+	}
+
+	err := fixtures.Clone(nodeAnalyzerFixtureDir, projects)
+	if err != nil {
+		panic(err)
+	}
+
+	err = initializeProjects(nodeAnalyzerFixtureDir)
+	if err != nil {
+		panic(err)
+	}
+
+	exitCode := m.Run()
+	defer os.Exit(exitCode)
+	// uncomment this if you need test files cleaned up locally
+	// defer cleanUp(nodeAnalyzerFixtureDir)
+}
+
+// While not testing the core functionality, this ensures that the tests have been setup correctly as needed for a prereq to run the analyzer steps
+// This test itself does not incur any overhead.
+func TestTestSetup(t *testing.T) {
+	t.Parallel()
+	assertProjectFixtureExists(t, "puppeteer")
+	// faker has no deps
+	// assertProjectFixtureExists(t, "fakerjs")
+	assertProjectFixtureExists(t, "fastify")
+	assertProjectFixtureExists(t, "nest")
+	assertProjectFixtureExists(t, "ohm")
+	assertProjectFixtureExists(t, "express")
+	assertProjectFixtureExists(t, "standard")
+	assertProjectFixtureExists(t, "sodium-encryption")
+	assertProjectFixtureExists(t, "request")
+}
+
+func TestAnalysisOutput(t *testing.T) {
+	t.Parallel()
+	for _, proj := range projects {
+		t.Run(proj.Name, func(t *testing.T) {
+			module := module.Module{
+				Dir:         filepath.Join(nodeAnalyzerFixtureDir, proj.Name),
+				Type:        pkg.NodeJS,
+				Name:        proj.Name,
+				Options:     map[string]interface{}{"allow-npm-err": true},
```

Comment:
We might want to use a table-driven test to set options only on modules that need it. I'd rather not set `allow-npm-err` for fixtures that don't need it.